lib/dependencies/HarmonyTopLevelThisParserPlugin.js

```diff
@@ -0,0 +1,23 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+"use strict";
+
+const ConstDependency = require("./ConstDependency");
+
+class HarmonyTopLevelThisParserPlugin {
+	apply(parser) {
+		parser.hooks.expression.for("this").tap("HarmonyTopLevelThisParserPlugin", node => {
+			const module = parser.state.module;
+			const isHarmony = !!(module.buildMeta && module.buildMeta.exportsType);
+			if(isHarmony && parser.scope.topLevelScope) {
+				const dep = new ConstDependency("undefined", node.range);
```

Comment:
I guess there is a 3rd argument to ConstDependency

Reply:
I don't need it right?

```js
constructor(expression, range, requireWebpackRequire) {
```

Reply:
yes you can pass `false`. The default is `true`